src/lib/Query/Content/CriterionVisitor/IsContainer.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Ibexa\Solr\Query\Content\CriterionVisitor;
+
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion\Operator;
+use Ibexa\Contracts\Solr\Query\CriterionVisitor;
+
+final class IsContainer extends CriterionVisitor
+{
+    public function canVisit(Criterion $criterion): bool
+    {
+        return $criterion instanceof Criterion\IsContainer && $criterion->operator === Operator::EQ;
+    }
+
+    public function visit(Criterion $criterion, CriterionVisitor $subVisitor = null): string
+    {
+        $value = $criterion->value;
+
+        if (!is_array($value) || !is_bool($value[0])) {
```

Comment:
Do we really accept array of values ?

Reply:
Currently, every scalar value is wrapped in an array here:
https://github.com/ibexa/core/blob/86d44648ace5e4ab48b9deae9e327145130113c9/src/contracts/Repository/Values/Content/Query/Criterion.php#L94
it was discussed with PHP and AFAICR we plan to change it in 5.0.